Dockerfiles/entrypoint.sh

```diff
@@ -1,23 +1,42 @@
 #!/bin/bash
 
 # operator-bundle-image validate
+validate_operator_bundle () {
+    echo -e "operatorSDKVersion is $operatorSDKVersion"
+    ansible-playbook -vvvv --connection local -i localhost, operator-test-playbooks/validate-operator-bundle.yml -e"work_dir=/project/output" -e"operator_work_dir=/project/operator-bundle/"
 
-ansible-playbook -vvvv --connection local -i localhost, operator-test-playbooks/validate-operator-bundle.yml -e"work_dir=/project/output" -e"operator_work_dir=/project/operator-bundle/"
+    operatorSDKVersion=$(cat /project/output/validation-version.txt)
+    cat /project/output/validation-output.txt
 
-operatorSDKVersion=$(cat /project/output/validation-version.txt)
-operatorBundlevalidationReturnCode=$(cat /project/output/validation-rc.txt)
-operatorBundlevalidationOutput=$(cat /project/output/validation-output.txt)
+    if [[ $? -eq 0 ]];
+    then
+      echo -e "\nBundle image validation is successful"
+      echo -e "Operator bundle validation Return Code is $operatorBundlevalidationReturnCode"
+      cat /project/output/validation-rc.txt
+      exit 0
+    else
+      echo -e "\nBundle image validation Failed"
+      exit 1
+    fi
+}
 
-echo -e "operatorSDKVersion is $operatorSDKVersion"
-echo -e "Operator bundle validation Return Code is $operatorBundlevalidationReturnCode"
-echo -e "Operator bundle Validation output is \n $operatorBundlevalidationOutput"
+#prepare-operator-metadata
+prepare_operator_metadata () {
+    ansible-playbook -vvv -i localhost, --connection local operator-test-playbooks/prepare-operator-metadata.yml -e "operator_dir=/project/operator_dir/" -e "operator_work_dir=/project/test_operator_work_dir/" -e"work_dir=/project/output/"
+    cat /project/output/parse_operator_metadata_results.json
+    cp /project/operator-test-playbooks/parse_operator_metadata_results.json /project/output
+}
 
-if [[ $operatorBundlevalidationReturnCode -eq 0 ]];
-then
-  echo -e "\nBundle image validation is successful"
-  exit 0
+TEST_NAME_ENV="${TEST_NAME:-validate_operator_bundle}"
+if [[ "$TEST_NAME_ENV" = 'validate_operator_bundle' ]]; then
+    echo "TEST_NAME is set to $TEST_NAME_ENV"
+    echo "Running operator bundle validation"
+    validate_operator_bundle
+elif [[ "$TEST_NAME_ENV" = 'prepare_operator_metadata' ]]; then
+    echo "TEST_NAME is set to $TEST_NAME_ENV"
+    echo "Running prepare_operator_metadata"
+    prepare_operator_metadata
 else
-  echo -e "\nBundle image validation Failed"
-  exit 1
+    echo "TEST_NAME is set to $TEST_NAME_ENV"
+    echo "$TEST_NAME_ENV not found in entrypoint.sh"
 fi
```

Comment:
Why do we change this file ?
It should be used just for midstream and it was not planned for the role roles/parse_operator_metadata

Reply:
I am making use of the same image without changing its functionality for testing. 
Do you think we need a separate image for testing? 
By default github actions run in ubuntu. 
Inorder to test it inside fedora am using podman to run inside fedora 33 container. 
Am using the same image to test our playbooks. 
Since the build time allocated to opensource projects is 1000 minutes for github actions afaik 
I am keeping it as minimum as possible.

Reply:
> Do you think we need a separate image for testing?

yes; midstream image should be used just for its purpose. Testing in midstream and it should be as small as possible.


> Since the build time allocated to opensource projects is 1000 minutes for github actions afaik
16 minutes is enough for unit-tests (including building the image)
Unit test have to be fast.